### DISCUSSION TITLE
Correlating gene expression with peak accessibility

### DISCUSSION THREAD

**Discussion by BartTheeuwes** [GreenleafLab/ArchR#1350] [Feature Requests]
Hi,

Thank you for this fantastic tool. 
I would be interested in finding potential target genes of a TF of interest. Currently, ArchR can already be used to detect which peaks contain a TF motif (using addMotifAnnotations), and to determine if these peaks are linked to a gene (peak2genelinks). However, I would be interested in determining if the expression (genescore/geneintegration) of the TF actually correlates with the accessibility those peaks, in order to subset the number of peaks that are likely to be bound.

I think what would be needed here is similar to the code for the peak2genelinks function, with the adaptation that it can be ran on a subset of the peaks/genes, and that it does not look at the genomic context (maxDist). 
Is this something that could be implemented?

Thanks

**Comment by rcorces** [comment]
Thanks for using ArchR. We dont have a function to do this and feature requests are quite slow to be implemented these days. However, this should be relatively straight-forward from an end-user perspective for a single peak (extract relevant rows from matrices using `getMatrixFromProject()` and correlate).

Alternatively, you can use the positive TF regulator workflow, though this isnt for a single peak. https://www.archrproject.com/bookdown/identification-of-positive-tf-regulators.html

**Reply by BartTheeuwes** [reply]
Thanks for the quick response. 
Correlating expression with accessibility will hopefully do the trick.

Best

**Comment by rojinsafavi** [comment]
Hi @rcorces , I have a follow-up question, let's say that E1 is the inferred expression of TF X, and P1 is the is a peak. Do you suggest performing the correlation analysis as follows? 

```cor.test(E1, P1)```

Or performing the correlation in pseudobulk?

The reason I'm asking this is that the correlation values that I'm getting are pretty low. 

Thank you!

**Reply by rcorces** [reply]
Seems reasonable to me. Though I'm not surprised to hear that the correlation values are low given the sparsity of the scATAC data. This is why most of the correlations in ArchR are performed on these lowly-overlapping cell groups (described in the paper methods).
